portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/UserMgtClientServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+
+        this.realmService = null;
+    }
+
+    @Override
+    public List<User> listUsers(int offset, int length, String domainName) throws UserPortalUIException {
+
+        List<User> userList;
+        try {
+            userList = getRealmService().getIdentityStore().listUsers(offset, length, domainName);
+        } catch (IdentityStoreException e) {
+            String error = "Error while listing users";
+            LOGGER.error(error, e);
+            throw new UserPortalUIException(error);
+        }
+        return userList;
+    }
+
+    @Override
+    public List<User> listUsers(Claim claim, int offset, int length, String domainName) throws UserPortalUIException {
+        List<User> userList;
+        try {
+            if (claim != null) {
+                userList = getRealmService().getIdentityStore().listUsers(claim, offset, length, domainName);
+            } else {
+                userList = this.listUsers(offset, length, domainName);
+            }
+        } catch (IdentityStoreException e) {
+            String error = "Error while listing users for claim " + claim.getClaimUri();
+            LOGGER.error(error, e);
+            throw new UserPortalUIException(error);
+        }
+        return userList;
+    }
+
+    @Override
+    public List<Claim> getClaimsOfUser(String uniqueUserId) throws UserPortalUIException {
+        List<Claim> userClaims;
+        try {
+            userClaims = getRealmService().getIdentityStore().getClaimsOfUser(uniqueUserId);
+        } catch (UserNotFoundException e) {
+            String error = "User not found with unique Id :  " + uniqueUserId;
+            LOGGER.error(error, e);
+            throw new UserPortalUIException(error);
+        } catch (IdentityStoreException e) {
+            String error = "Error while getting claims of user with unique Id :  " + uniqueUserId;
+            LOGGER.error(error, e);
+            throw new UserPortalUIException(error);
+        }
+        return userClaims;
+    }
+
+    @Override
+    public List<UserUIEntry> getUsersForList(int offset, int length, String domainName,
+                                             String usernameClaim) throws UserPortalUIException {
+        List<UserUIEntry> userList = new ArrayList<>();
+        List<User> users = new ArrayList<>();
+        try {
+            users = getRealmService().getIdentityStore().listUsers(offset, length, domainName);
+        } catch (IdentityStoreException e) {
+
+        }
+
+        List<MetaClaim> metaClaims = new ArrayList<>();
+        MetaClaim metaClaim = new MetaClaim();
+        metaClaim.setClaimUri(usernameClaim);
+        metaClaims.add(metaClaim);
+
+        for (User user : users) {
+            List<Group> groups;
+            List<Claim> userIds;
+            String username = null;
+            try {
+                groups = user.getGroups();
+                userIds = user.getClaims(metaClaims);
+                if (!userIds.isEmpty()) {
```

Comment:
It will not show any detail in Username column of the list table for that user. if the profile is not defined, http://wso2.org/claims/username claim value will be in userId. list will be empty only if the profile is not set and http://wso2.org/claims/username doesn't contain any value.